components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/client/authentication/OAuthClientAuthenticator.java

```diff
@@ -65,4 +66,14 @@ boolean canAuthenticate(HttpServletRequest request, Map<String, List> bodyParams
      */
     String getClientId(HttpServletRequest request, Map<String, List> bodyParams, OAuthClientAuthnContext
             oAuthClientAuthnContext) throws OAuthClientAuthnException;
+
+    /**
+     * Retrieve the authentication methods supported by the authenticator.
+     *
+     * @return      Authentication methods supported by the authenticator.
+     */
+    default List<String> getSupportedClientAuthenticationMethods() {
+
+            return new ArrayList<>();
```

Comment:
```suggestion
            return Collections.emptyList();
```

Reply:
Fixed in 428fe44b8129a99fb54bc22701c96ca5c6dc0bff